### PR TITLE
Handle passports for certification requests

### PR DESCRIPTION
**Pull request by Remg**

### CODE REVIEW

File: src/Vision/IdentityDocumentParser.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        throw new \InvalidArgumentException(sprintf('Instance of "%s" is not a handled identity document type.', ImageAnnotations::class));
+    }
+
+    public function hasLastName(ImageAnnotations $imageAnnotations, string $lastName): bool
+    {
+        if ($imageAnnotations->isFrenchNationalIdentityCard()) {
+            preg_match('/Nom( )?:( )?(?<last_name>.+)\\n/', $this->text, $matches);
+
+            dump($this->normalize($lastName));
+            dump($matches['last_name'] ?? null);
```
Comment (kevin-verschaeve) [inline src/Vision/IdentityDocumentParser.php:50]: oops

File: src/Vision/IdentityDocumentParser.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return $this->hasFirstName($imageAnnotations, $adherent->getFirstName())
+            && $this->hasLastName($imageAnnotations, $adherent->getLastName())
+            && $this->hasDateOfBirth($imageAnnotations, $adherent->getBirthdate())
+        ;
+    }
+
+    public function hasFirstName(ImageAnnotations $imageAnnotations, string $firstName): bool
+    {
+        if ($imageAnnotations->isFrenchNationalIdentityCard()) {
+            preg_match('/Prénom( )?\(s\)( )?:( )?(?<first_names>.+)\\n/', $imageAnnotations->getText(), $matches);
```
Comment (kevin-verschaeve) [inline src/Vision/IdentityDocumentParser.php:28]: I had updated this regex in my previous PR to fix an issue with parentheses. Don't you want to reuse it ?

File: src/Vision/IdentityDocumentParser.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                return false;
+            }
+
+            $birthDate = \DateTime::createFromFormat('d.m.Y', str_replace(' ', '.', $matches['birth_date']));
+
+            return $birthDate->format('Y-m-d') === $dateOfBirth->format('Y-m-d');
+        } elseif ($imageAnnotations->isFrenchPassport()) {
+            $payload = $this->normalize($imageAnnotations->getText());
+
+            $pattern = sprintf('/(?<birth_date>%s(\-)?%s(\-)?%s)/',
```
Comment (kevin-verschaeve) [inline src/Vision/IdentityDocumentParser.php:77]: I think parentheses and escape char are not mandatory here
```suggestion
            $pattern = sprintf('/(?<birth_date>%s-?%s-?%s)/',
```

File: src/Vision/IdentityDocumentParser.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            return false !== strpos($payload, $this->normalize($lastName));
+        }
+
+        throw new \InvalidArgumentException(sprintf('Instance of "%s" is not a handled identity document type.', ImageAnnotations::class));
+    }
+
+    public function hasDateOfBirth(ImageAnnotations $imageAnnotations, \DateTimeInterface $dateOfBirth): bool
+    {
+        if ($imageAnnotations->isFrenchNationalIdentityCard()) {
+            preg_match('/(?<birth_date>[\d]{2}[\. ][\d]{2}[\. ][\d]{4})/', $this->text, $matches);
```
Comment (kevin-verschaeve) [inline src/Vision/IdentityDocumentParser.php:65]: You can remove brakcets when there is only one allowed value
```suggestion
            preg_match('/(?<birth_date>\d{2}[\. ]\d{2}[\. ]\d{4})/', $this->text, $matches);
```

File: src/Vision/IdentityDocumentParser.php (hunk 1 above)
Comment (Remg) [reply src/Vision/IdentityDocumentParser.php]: i have no idea what you're talking about

**Review by ottaviano — APPROVED** [review]

**Review by kevin-verschaeve — APPROVED** [review]

**Pull request merged**